x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/local/ReplaceMissingFieldWithNull.java

```diff
@@ -81,7 +81,9 @@ else if (plan instanceof Project project) {
                     Alias nullAlias = nullLiteral.get(f.dataType());
                     // save the first field as null (per datatype)
                     if (nullAlias == null) {
-                        Alias alias = new Alias(f.source(), f.name(), Literal.of(f, null), f.id());
+                        Alias alias = joinAttributes.isEmpty()
+                            ? new Alias(f.source(), f.name(), Literal.of(f, null), f.id())
+                            : new Alias(f.source(), f.name(), Literal.of(f, null));
```

Comment:
1. please add an explanation on why this occurs
2. use the ternary operator for the id to pick either f.id or null and call the constructor only once
`new Alias(f.source(), f.name(), Literal.of(), joinAttributes.isEmpty() ? f.id(): null))`

As a follow-up the fact that the rule has to be aware of the join is fishy.